mlir/lib/Dialect/Vector/Transforms/VectorTransforms.cpp

```diff
@@ -1514,6 +1625,15 @@ void mlir::vector::populateShapeCastFoldingPatterns(RewritePatternSet &patterns,
   patterns.add<ShapeCastOpFolder>(patterns.getContext(), benefit);
 }
 
+void mlir::vector::populateReorderShapeCastPatterns(RewritePatternSet &patterns,
+                                                    PatternBenefit benefit) {
+  patterns.add<ReorderShapeCastWithUnitDimAndArith<arith::AddIOp>,
+               ReorderShapeCastWithUnitDimAndArith<arith::AddFOp>,
+               ReorderShapeCastWithUnitDimAndArith<arith::MulIOp>,
+               ReorderShapeCastWithUnitDimAndArith<arith::MulFOp>>(
+      patterns.getContext(), benefit);
```

Comment:
We should also add the shape_cast canonicalization patterns here.

Reply:
Added `ShapeCastOpFolder` 👍🏻